catalogue/webapp/components/WorkDetails/WorkDetails.tsx

```diff
@@ -591,7 +591,26 @@ const WorkDetails: FunctionComponent<Props> = ({ work }: Props) => {
             text={[work.physicalDescription]}
           />
         )}
-
+        {work.partOf.filter(p => !p['id']).length > 0 && (
```

Comment:
`some` short-circuits when it finds something which satisfies the predicate, although if you put your `work.partOf.filter(...)` in a variable somewhere it'd save you looping over the same stuff again a few lines down...

```suggestion
        {work.partOf.some(p => !['id']) && (
```